### PR TITLE
SONARJAVA-1117 FP in S109 (MagicNumber): Numbers used in constructor of enum constants should be ignored

### PR DESCRIPTION
**Pull request by merciesa**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/MagicNumberCheck.java [hunk 1]
```diff
@@ -57,6 +58,11 @@ public void scanFile(JavaFileScannerContext context) {
   }
 
   @Override
+  public void visitEnumConstant(EnumConstantTree tree) {
+    // Ignore literals within enum constant
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/MagicNumberCheck.java]: and what about a case like : 
```
enum Foo {
  PLOP(100) {
   int i = 100;
  void foo(int i) {
    System.out.println(42);
  }
};
void foo(int i) {}
}
```

Comment (merciesa) [inline java-checks/src/main/java/org/sonar/java/checks/MagicNumberCheck.java]: Do you see a case where the initializer of the enum could be something else than a NewClassTree ?

Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/MagicNumberCheck.java]: Check in the construction of the EnumConstantTreeImpl : https://github.com/SonarSource/sonar-java/blob/master/java-squid/src/main/java/org/sonar/java/model/declaration/EnumConstantTreeImpl.java#L32

File: java-checks/src/main/java/org/sonar/java/checks/MagicNumberCheck.java [hunk 2]
```diff
@@ -57,6 +59,11 @@ public void scanFile(JavaFileScannerContext context) {
   }
 
   @Override
+  public void visitEnumConstant(EnumConstantTree tree) {
+    scan(((NewClassTree) tree.initializer()).classBody());
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/MagicNumberCheck.java]: I created ticket and fixed it : https://github.com/SonarSource/sonar-java/pull/291

**Comment by benzonico** [comment]
LGTM. Can you wait until https://github.com/SonarSource/sonar-java/pull/291 is merged so you can actually remove the useless cast.

**Pull request merged**